pkg/config/piped.go

```diff
@@ -181,24 +184,38 @@ func (s *PipedSpec) EnableDefaultKubernetesCloudProvider() {
 	s.CloudProviders = append(s.CloudProviders, defaultKubernetesPlatformProvider)
 }
 
-// HasCloudProvider checks whether the given provider is configured or not.
-func (s *PipedSpec) HasCloudProvider(name string, t model.ApplicationKind) bool {
-	requiredProviderType := t.CompatibleCloudProviderType()
-	for _, cp := range s.CloudProviders {
-		if cp.Name != name {
-			continue
+// HasPlatformProvider checks whether the given provider is configured or not.
+// Note: Currently check both list CloudProviders & PlatformProviders for configuration but we
+// will disable CloudProviders support next few versions.
+func (s *PipedSpec) HasPlatformProvider(name string, t model.ApplicationKind) bool {
+	requiredProviderType := t.CompatiblePlatformProviderType()
+	for _, cp := range s.PlatformProviders {
+		if cp.Name == name && cp.Type == requiredProviderType {
+			return true
 		}
-		if cp.Type != requiredProviderType {
-			continue
+	}
+	for _, pp := range s.CloudProviders {
+		if pp.Name == name && pp.Type == requiredProviderType {
+			return true
 		}
-		return true
 	}
 	return false
 }
```

Comment:
To avoid duplication how about removing this function and use `FindCloudProvider` instead?